src/device/mod.rs

```diff
@@ -1786,6 +1787,34 @@ impl TaskRuntime for Runtime {
                 Ok(())
             },
 
+            Some(pq_event) = self.event_listeners.post_quantum.recv() => {
+                telio_log_debug!("WG consolidation triggered by PQ event");
+
+                match pq_event {
+                    pq::Event::Handshake(keys) => {
+                        self.requested_state.postquantum_keys = Some(keys);
+                    },
+                    pq::Event::Rekey(pq::Keys { wg_secret, pq_shared }) => {
+                        if let Some(keys) = &mut self.requested_state.postquantum_keys {
+                            // Check if we are still talking to the same exit node
+                            if keys.wg_secret == wg_secret {
+                                // and only then update the preshared key,
+                                // otherwise we're connecting to different node already
+                                keys.pq_shared = pq_shared;
+                            }
+                        }
+                    },
+                }
```

Comment:
This is a solid approach but I think adding a logging statement of low severity might aid in debuggin if the keys don't match.

Reply:
Done